From 82c2c799b3027ec25b4c53cd836ff4be3fb74a1b Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Tue, 16 Jan 2018 20:28:48 +0100
Subject: [PATCH] TOOLS-1858 Want sdcadm updates channel to set based on
 installer Reviewed by: Angela Fong <angela.fong@joyent.com> Reviewed by:
 Trent Mick <trentm@gmail.com> Approved by: Angela Fong
 <angela.fong@joyent.com> Approved by: Trent Mick <trentm@gmail.com>

---
 answers.json.tmpl          |  3 ++-
 answers.json.tmpl.external |  3 ++-
 scripts/prompt-config.sh   | 32 ++++++++++++++++++++++++++++++++
 3 files changed, 36 insertions(+), 2 deletions(-)

diff --git a/answers.json.tmpl b/answers.json.tmpl
index 7656e1af..b264a920 100644
--- a/answers.json.tmpl
+++ b/answers.json.tmpl
@@ -27,5 +27,6 @@
     "api_password": "joypass123",
     "mail_to": "<default>",
     "mail_from": "<default>",
-    "phonehome_automatic": "<default>"
+    "phonehome_automatic": "<default>",
+    "update_channel": "release"
 }
diff --git a/answers.json.tmpl.external b/answers.json.tmpl.external
index 38f495c1..aac46362 100644
--- a/answers.json.tmpl.external
+++ b/answers.json.tmpl.external
@@ -33,5 +33,6 @@
     "api_password": "joypass123",
     "mail_to": "<default>",
     "mail_from": "<default>",
-    "phonehome_automatic": "<default>"
+    "phonehome_automatic": "<default>",
+    "update_channel": "release"
 }
diff --git a/scripts/prompt-config.sh b/scripts/prompt-config.sh
index 6751830a..99a87ee1 100755
--- a/scripts/prompt-config.sh
+++ b/scripts/prompt-config.sh
@@ -1199,6 +1199,34 @@ Joyent to help us make SmartDataCenter better.
 	phonehome_automatic="$val"
 
 
+	printheader "Update Channel"
+	message="
+Channel to be used for software update. Options:
+- support: supported versions, intended for customers with Joyent support
+  contracts
+- release: biweekly release versions
+- dev: development builds
+\n"
+
+	if [[ $(getanswer "skip_instructions") != "true" ]]; then
+		printf "$message"
+	fi
+
+	while [ true ]; do
+		key="update_channel"
+		promptval "Enter the update channel" \
+			"$update_channel" "${key}"
+		if [[ "$val" == "dev" || "$val" == "release" || \
+					"$val" == "support" ]]; then
+			update_channel="$val"
+			break
+		fi
+		echo "Must select one of 'support', 'release' or 'dev'"
+		# disable key, since this means bad value in answer file
+		key=
+	done
+
+
 	printheader "Verify Configuration"
 	message=""
 
@@ -1244,6 +1272,8 @@ Joyent to help us make SmartDataCenter better.
 		echo
 		printf "Enable telemetry: $phonehome_automatic\n"
 		echo
+		printf "Update channel: $update_channel\n"
+		echo
 	fi
 
 	if [[ $(getanswer "skip_final_confirm") != "true" ]]; then
@@ -1660,6 +1690,8 @@ echo >>$tmp_config
 
 echo "phonehome_automatic=${phonehome_automatic}" >>$tmp_config
 
+echo "update_channel=${update_channel}" >>$tmp_config
+
 # Always show the timers and make setup serial for now.
 echo "show_setup_timers=true" >> $tmp_config
 if [[ $(getanswer "dtrace_zone_setup") == "true" ]]; then
-- 
2.21.0

